stack/core/src/main/java/org/apache/usergrid/corepersistence/CpWalker.java

```diff
@@ -58,8 +58,10 @@ public CpWalker(){
     public void walkCollections(final CpEntityManager em, final EntityRef start,
         String collectionName, boolean reverse, final CpVisitor visitor) throws Exception {
 
-        doWalkCollections(
-            em, collectionName, reverse, new SimpleId( start.getUuid(), start.getType() ), visitor );
+        if(start != null) {
```

Comment:
Should we fail silently or blow up here?  Feels like we should do preconditions so we can flush out the incorrect caller impl